src/NBitcoin/BitcoinCore/Coins.cs

```diff
@@ -160,7 +160,7 @@ public void ReadWrite(BitcoinStream stream)
                 // coinbase height
                 stream.ReadWriteAsVarInt(ref this.nHeight);
 
-                if (stream.ConsensusFactory.Consensus.IsProofOfStake)
+                if (stream.ConsensusFactory.Consensus != null && stream.ConsensusFactory.Consensus.IsProofOfStake)
```

Comment:
Ah interesting this should throw DefaultConsensusFactory should throw is a stream is trying to access `DefaultConsensusFactory.Consensus` is my opinion.

Reply:
OK yeah I've removed this null check 👍 . Please see latest commit